src/backup_sram.cpp

```diff
@@ -5,39 +5,66 @@
 
 void backup_sram_init()
 {
+  //The checksum function depends on the BackupData struct being a multiple of 32 bits
+  //If you change the BackupData struct, and it trips on this static_assert,
+  //either make the BackupData struct a multiple of 32 bits, or change the checksum method
+  static_assert(sizeof(rosflight_firmware::BackupData)%4==0,"Error with debug_info_t: Size is not multiple of 32 bits");
 #pragma GCC diagnostic push //Ignore old style cast from included library
 #pragma GCC diagnostic ignored "-Wold-style-cast"
-    RCC_APB1PeriphResetCmd(RCC_APB1Periph_PWR, DISABLE);
-    RCC_APB1PeriphClockCmd(RCC_APB1Periph_PWR, ENABLE);
-    RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_BKPSRAM, ENABLE);
+  RCC_APB1PeriphResetCmd(RCC_APB1Periph_PWR, DISABLE);
+  RCC_APB1PeriphClockCmd(RCC_APB1Periph_PWR, ENABLE);
+  RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_BKPSRAM, ENABLE);
 #pragma GCC diagnostic pop
-    PWR_BackupRegulatorCmd(ENABLE);
+  PWR_BackupRegulatorCmd(ENABLE);
 }
 
-void backup_sram_write(const void *src, size_t len)
+void backup_sram_write(const rosflight_firmware::BackupData &data)
 {
-    PWR_BackupAccessCmd(ENABLE);
+  PWR_BackupAccessCmd(ENABLE);
 #pragma GCC diagnostic push //Ignore old style cast from included library
 #pragma GCC diagnostic ignored "-Wold-style-cast"
```

Comment:
would it make sense to push all the pragmas to the outside of these functions?  I think it would make this a lot more legible